CMakeLists.txt

```diff
@@ -241,6 +241,16 @@ if( TRACCC_SETUP_ALPAKA )
    # Default options for the Alpaka build.
    set( alpaka_ACC_CPU_B_SEQ_T_THREADS_ENABLE TRUE CACHE BOOL
       "Enable the serial backend of Alpaka" )
+   #enable_language cannot be called by a function: put it in the top-level directory so it's available for all client code
+   if(alpaka_ACC_GPU_CUDA_ENABLE)
+     enable_language(CUDA)
+     include( traccc-compiler-options-cuda )
+   elseif(alpaka_ACC_GPU_HIP_ENABLE)
+     enable_language(HIP)
+   elseif(alpaka_ACC_SYCL_ENABLE)
+     enable_language(SYCL)
+     include( traccc-compiler-options-sycl )
+   endif()
```

Comment:
I don't like this. :thinking: This way we set our own specific flags for all the externals as well. The reason that we don't include `traccc-compile-options-cpp` in the main file is not to pollute the settings of the externals.

You already have these blocks in the appropriate places. If you don't want to have them in 3 different places, then introduce a macro for doing this.

https://cmake.org/cmake/help/latest/command/macro.html

As the name suggests, macros run in the scope in which they are launched. So any variable set inside of a macro, "leaks" into the caller's scope. (Which in this case is what we'd want.)